### PR TITLE
Move role verificator to auth module

### PR DESCRIPTION
**Pull request by dominik003**
This PR moves RoleVerficiation and ProjectRoleVerification from the database submodule of authentication to injectables.py. In addition, the current import uses of the two classes have been adapted to the correct format. Finally, the unused deprecated functions have been removed.

### CODE REVIEW

File: backend/capellacollab/core/authentication/injectables.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    def __call__(
+        self,
+        token=fastapi.Depends(jwt_bearer.JWTBearer()),
+        db=fastapi.Depends(database.get_db),
+    ) -> bool:
+        role = users_crud.get_user_by_name(
+            db=db, username=helper.get_username(token)
+        ).role
+        if (
+            role == users_models.Role.USER
```
Comment (amolenaar) [inline backend/capellacollab/core/authentication/injectables.py:33]:
```suggestion
            role != users_models.Role.ADMIN
```

With this statement I do not have to know there are only 2 roles in the system.

File: backend/capellacollab/core/authentication/injectables.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+        project = projects_crud.get_project_by_slug(db, project_slug)
+        project_user = projects_users_crud.get_user_of_project(
+            db, project, user
+        )
+
+        # Check role
+        if not project_user or self.roles.index(
+            project_user.role
+        ) < self.roles.index(self.required_role):
```
Comment (amolenaar) [inline backend/capellacollab/core/authentication/injectables.py:85]: ... or use a line similar to this in `RoleVerification.__call__` too.

**Review by amolenaar — APPROVED** [review]
One small improvement regarding roles. I think the code can be decoupled from the actual role.

File: backend/capellacollab/core/authentication/injectables.py (hunk 1 above)
Comment (dominik003) [reply backend/capellacollab/core/authentication/injectables.py:33]: Well spotted. I think this is sufficient for now, as it at least prevents the case where we add another user role that would then have come through this check.

File: backend/capellacollab/core/authentication/injectables.py (hunk 2 above)
Comment (dominik003) [reply backend/capellacollab/core/authentication/injectables.py:85]: Applied your suggestion from above.

**Pull request merged**